### PR TITLE
Helpful message when missing parent context

### PR DESCRIPTION
**Pull request by Dtphelan1**
Supports a help message when a shortcut is incomplete because a parent shortcut is missing. Additionally had to make some other changes in the form of refactoring and code additions.:
- Cleaned up the getHelp completion component a bit. 
- Pass shortcutManager to completionComponents now.
- Added labels to any shortcuts that show up as KnownParentContexts. These labels, in addition to being used as a placeholder for the shortcut when being expanded (often not a problem for any of the shortcuts touched here), are a form of readable text to present to the user. Thus we use shortcut.label when trying to figure out how to display something in the GetHelp portal. 
- CompletionPortalSearchIndices now use TriggersWithoutLabel to define their filtered list of shortcuts.
- Updated the TriggersWithoutLabel function to remove one instance of a label in thelist of triggers instead of all instances. This avoids issues with shortcuts like @condition, where the label is identical to one of its string triggers. 

To test basic functionality: 
- Insert #disease status into the editor, and open the getHelp menu. Confirm that the message displayed is sufficiently helpful. 

To test fully...
- I think we need to give a test on all the shortcuts we have. I tried to test this as much as I could but I'm worried I might have missed a few cases. 

_Pull requests into Flux require the following. Submitter should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:. Reviewers should complete every item on the bulleted list when reviewing._

## Submitter:

**Running the Application**

- [ ] Manually tested in Chrome (primary browser for testing)
- [ ] Manually tested in IE/Safari (verify app loads and basic feature testing)
- [ ] Manually tested in Chrome in iPad mode (at minimum, viewing patient record)

**Task**

- [ ] Branch implements task as expected
- [ ] Key application features are updated as needed for task and tested (see: [Key Application Features](https://github.com/FluxNotes/flux/wiki/Key-Application-Features))
- [ ] Task specific definition of done (as described in the Jira task) is met, if applicable.

**Documentation**

- [ ] This pull request describes why these changes were made
- [ ] Recognizes any potential shortcomings/bugs in the description 
- [ ] Pre release script is updated on the [Pre Release Testing Script wiki page](https://github.com/FluxNotes/flux/wiki/Pre-Release-Testing-Script) (add/remove single features to test, do not run through full script)
- [ ] Update the list of key features with new features on the [Key Application Features wiki page](https://github.com/FluxNotes/flux/wiki/Key-Application-Features)
- [ ] Document any new APIs/extension points
- [ ] Additional technical components and libraries are documented and added to [wiki](https://github.com/FluxNotes/flux/wiki/About-Flux-Notes) as needed

**Code Quality**

- [ ] Code style guide is followed (see: [Code Style Guide](https://github.com/FluxNotes/flux/wiki/JavaScript-and-HTML-Code-Style-Guide))
- [ ] Code is commented

**Tests**

- [ ] Existing tests passed
- [ ] Added backend tests for new functionality added

## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code

### CODE REVIEW

**Comment by sgober** [comment]
[SO1] Some weird behavior with #date... if the parent context (#as of, #reference date) is missing there is no way to indicate what is missing because the date picker opens.  I have no idea how to get around this so this might just be an observation

**Comment by sgober** [comment]
@Dtphelan1 I agree, I have no idea how to handle this scenario unless we can somehow add a note at the top or bottom of the datepicker.  Let's make a JIRA! back to reviewing...

**Comment by mgramigna** [comment]
[MG1?] I noticed we don't have a portal pop up for inserters that require a parent context, but don't require value selection from a portal, like @recent labs and @oncohist.

So if I enter @recent labs on its own (which seems like not too crazy of a possibility), I won't know that I need to add @condition before it to complete it, like I would be told to do with disease status

**Comment by mgramigna** [comment]
Re: MG1

It looks like we only added the potential parents logic to creator bases and intermediaries?

**Comment by Dtphelan1** [comment]
So these are all good considerations. I think even if we added the ability to solve for those values on the insertValue classes, we wouldn't be able to use those values to display information in the form of the GetHelp menu since these components don't have a getHelp component to display. My thinking is that we nest this issue under the same JIRA associated with @sgober 's comment above. Thoughts?

File: src/shortcuts/CreatorIntermediary.jsx [hunk 1]
```diff
@@ -115,10 +115,27 @@ export default class CreatorIntermediary extends Shortcut {
...
         return this.hasParentContext() && this.hasChildren();
     }
+
+    get isMissingParent() {
+        return !this.hasParentContext();
+    }
+    get potentialParents() {
+        const knownParent = this.metadata["knownParentContexts"];
+        if (knownParent === 'Patient' || knownParent === undefined) return [];
+        console.log(knownParent);
```
[the remark this answers is not in the document]
Comment (Dtphelan1) [reply src/shortcuts/CreatorIntermediary.jsx]: boi

**Comment by Dtphelan1** [comment]
Addressed MG2 - let me know when you can rereview @mgramigna

**Comment by mgramigna** [comment]
before 1pm ideally

**Pull request merged**